nixos/modules/services/development/hoogle.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+
+# services.hoogle = {
+#   enable = true;
+#   packages = hp: with hp; [ text lens ];
+#   haskellPackages = pkgs.haskellPackages;
+# };
+
+with lib;
+
+let
+
+  cfg = config.services.hoogle;
+  ghcWithHoogle = pkgs.haskellPackages.ghcWithHoogle;
+
+in {
+
+  options.services.hoogle = {
+    enable = mkEnableOption "Hoogle Documentation service";
+
+    port = mkOption {
+      type = types.int;
+      default = 8080;
+      description = ''
+        Port number Hoogle will be listening to.
+      '';
+    };
+
+    packages = mkOption {
+      default = hp: [];
+      example = "hp: with hp; [ text lens ]";
+      description = ''
+        A function that takes a haskell package set and returns a list of
+        packages from it.
+      '';
+    };
+
+    haskellPackages = mkOption {
```

Comment:
This option ought to have a default; you can do
`default = pkgs.haskellPackages; defaultText = "pkgs.haskellPackages"`

In the example, perhaps show off using a different package set than the default (i.e., for a specific compiler version).